src/sdk/helpers/smil-document-parser.js

```diff
@@ -0,0 +1,248 @@
+                    parsingDeferred.media_overlay_id = spineItem.media_overlay_id;
+                    parsingDeferreds.push(parsingDeferred);
+                    var smilJson = {};
+
+                    // Push the holder object onto the map early so that order isn't disturbed by asynchronicity:
+                    mo_map.push(smilJson);
+
+                    // The local parsingDeferred variable will have its value replaced on next loop iteration.
+                    // Must pass the parsingDeferred through async calls as an argument and it arrives back as myDeferred.
+                    that.parse(spineItem, manifestItemSMIL, smilJson, parsingDeferred, function(myDeferred, smilJson) {
+                        allFakeSmil = false;
+                        myDeferred.resolve();
+                    }, function(myDeferred, parseError) {
+                        console.log('Error when parsing SMIL manifest item ' + manifestItemSMIL.href + ':');
+                        console.log(parseError);
+                        myDeferred.resolve();
+                    });
+                } else {
+                    mo_map.push(makeFakeSmilJson(spineItem));
+                }
+            }
+
+            $.when.apply($, parsingDeferreds).done(function() {
+                packageDocument.getMetadata().setMoMap(mo_map);
+                if (allFakeSmil) {
+                    console.log("No Media Overlays");
+                    packageDocument.getMetadata().setMoMap([]);
+                }
+                callback();
+            });
+        }
+    };
+
+    // parse the timestamp and return the value in seconds
+    // supports this syntax:
+    // http://idpf.org/epub/30/spec/epub30-mediaoverlays.html#app-clock-examples
+    SmilDocumentParser.resolveClockValue = function(value) {
+        if (!value) return 0;
+
+        var hours = 0;
+        var mins = 0;
+        var secs = 0;
+
+        if (value.indexOf("min") != -1) {
+            mins = parseFloat(value.substr(0, value.indexOf("min")));
+        } else if (value.indexOf("ms") != -1) {
+            var ms = parseFloat(value.substr(0, value.indexOf("ms")));
+            secs = ms / 1000;
+        } else if (value.indexOf("s") != -1) {
+            secs = parseFloat(value.substr(0, value.indexOf("s")));
+        } else if (value.indexOf("h") != -1) {
+            hours = parseFloat(value.substr(0, value.indexOf("h")));
+        } else {
+            // parse as hh:mm:ss.fraction
+            // this also works for seconds-only, e.g. 12.345
+            var arr = value.split(":");
+            secs = parseFloat(arr.pop());
+            if (arr.length > 0) {
+                mins = parseFloat(arr.pop());
+                if (arr.length > 0) {
+                    hours = parseFloat(arr.pop());
+                }
+            }
+        }
+        var total = hours * 3600 + mins * 60 + secs;
+        return total;
+    }
+    
+    return SmilDocumentParser;
```

Comment:
We need to export `SmilDocumentParser` like `module.exports = SmilDocumentParser`. The wrapping function of `define` isn't needed.